File: recipes/scala/src/main/scala/com/adobe/platform/ml/Evaluator.scala
/*#####################################################################
# ADOBE CONFIDENTIAL
# ___________________
#
#  Copyright 2019 Adobe
#  All Rights Reserved.
#
# NOTICE:  All information contained herein is, and remains
# the property of Adobe and its suppliers, if any. The intellectual
# and technical concepts contained herein are proprietary to Adobe
# and its suppliers and are protected by all applicable intellectual
# property laws, including trade secret and copyright laws.
# Dissemination of this information or reproduction of this material
# is strictly forbidden unless prior written permission is obtained
# from Adobe.
#####################################################################*/

package com.adobe.platform.ml

import java.util

import com.adobe.platform.ml.impl.DefaultRegressionEvaluator
import com.adobe.platform.ml.sdk.MLMetric
import com.adobe.platform.ml.config.ConfigProperties
import org.apache.spark.ml.Transformer
import org.apache.spark.sql.{DataFrame, SparkSession}
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types.TimestampType

/**
  * Implementation of Evaluator which splits the dataframe and evaluates it
  */

class Evaluator extends DefaultRegressionEvaluator {

  /**
    *
    * @param configProperties - Configuration Properties map
    * @param data             - DataFrame that is used to split the data
    * @return                 - Tuple of the dataframes
    */

  override def split(configProperties:ConfigProperties, data: DataFrame): (DataFrame, DataFrame) = {

    val sparkSession = data.sparkSession
    import sparkSession.implicits._

    // Order by date and split the data
    var df = data.orderBy("date").withColumn("date", $"date".cast("String"))
    var train_data = df.filter($"date".<=("2012-02-10 00:00:00"))
    var test_data = df.filter($"date".>("2012-01-27 00:00:00"))
    (train_data, test_data)
  }

}

File: recipes/feature_pipeline_recipes/pyspark/pysparkfeaturepipelineapp/evaluator.py
#####################################################################
# ADOBE CONFIDENTIAL
# ___________________
#
#  Copyright 2020 Adobe
#  All Rights Reserved.
#
# NOTICE:  All information contained herein is, and remains
# the property of Adobe and its suppliers, if any. The intellectual
# and technical concepts contained herein are proprietary to Adobe
# and its suppliers and are protected by all applicable intellectual
# property laws, including trade secret and copyright laws.
# Dissemination of this information or reproduction of this material
# is strictly forbidden unless prior written permission is obtained
# from Adobe.
#####################################################################

from pyspark.sql.functions import unix_timestamp, from_unixtime, to_date, lit, col
from sdk.evaluation.regression import RegressionEvaluator
from .helper import setupLogger


class Evaluator(RegressionEvaluator):
    logger = setupLogger(__name__)

    def __init__(self):
        print("Initiate")

    def split(self, config_properties, dataframe):
        # Order by date and split the data
        train = dataframe.filter(dataframe["date"] <= lit('2012-01-27')).orderBy("date")
        test = dataframe.filter(dataframe["date"] > lit('2012-01-27')).orderBy("date")
        self.logger.debug("Training dataset count is %s " % train.count())
        self.logger.debug("Testing dataset count is %s " % test.count())
        return train, test

